### ISSUE TITLE
[tune.logger] Could not instantiate TFLogger with tensorflow 2.0.0 beta

### ISSUE THREAD

**Issue by nlnjnj** [ray-project/ray#5502]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 19.04
- **Ray installed from (source or binary)**: source
- **Ray version**: 0.8.0.dev3
- **Python version**: 3.7
- **Exact command to reproduce**:

### Describe the problem

![code](https://user-images.githubusercontent.com/5831940/63484178-5c76a880-c4d1-11e9-95cf-be46eab14bac.png)

As the IDE marked, this issue is caused by API version and so on.

I just like to know if there is any plan for the tensorflow 2.0.0 support?

### Source code / logs

2019-08-22 03:29:58,409	WARNING logger.py:255 -- Could not instantiate <class 'ray.tune.logger.TFLogger'> - skipping.

**Comment by richardliaw** [comment]
There's an example TF2 Logger (thanks @morgangiraud):

```
class TF2Logger(Logger):
    def _init(self):
        self._file_writer = tf.summary.create_file_writer(self.logdir)

    def on_result(self, result):
        with tf.device('/CPU:0'):
            with self._file_writer.as_default():
                step = result.get(
                    TIMESTEPS_TOTAL) or result[TRAINING_ITERATION]

                tmp = result.copy()
                for k in [
                        "config", "pid", "timestamp", TIME_TOTAL_S,
                        TRAINING_ITERATION
                ]:
                    if k in tmp:
                        del tmp[k]  # not useful to log these

                for attr, value in tmp.items():
                    if type(value) in [int, float]:
                        tf.summary.scalar(attr, value, step=step)
        self._file_writer.flush()

    def flush(self):
        self._file_writer.flush()

    def close(self):
        self._file_writer.close()
```
You can use this and replace the standard Tune TF Logger.

**Comment by nlnjnj** [comment]
> There's an example TF2 Logger (thanks @morgangiraud):
> 
> ```
> class TF2Logger(Logger):
>     def _init(self):
>         self._file_writer = tf.summary.create_file_writer(self.logdir)
> 
>     def on_result(self, result):
>         with tf.device('/CPU:0'):
>             with self._file_writer.as_default():
>                 step = result.get(
>                     TIMESTEPS_TOTAL) or result[TRAINING_ITERATION]
> 
>                 tmp = result.copy()
>                 for k in [
>                         "config", "pid", "timestamp", TIME_TOTAL_S,
>                         TRAINING_ITERATION
>                 ]:
>                     if k in tmp:
>                         del tmp[k]  # not useful to log these
> 
>                 for attr, value in tmp.items():
>                     if type(value) in [int, float]:
>                         tf.summary.scalar(attr, value, step=step)
>         self._file_writer.flush()
> 
>     def flush(self):
>         self._file_writer.flush()
> 
>     def close(self):
>         self._file_writer.close()
> ```
> 
> You can use this and replace the standard Tune TF Logger.

Thanks, I suggest if we could use tensorboard standalone for logger purpose since pytorch only relies on tensorboard 1.14 or above

**Comment by richardliaw** [comment]
Yeah it should be totally possible; the WARNING message is non-fatal, and you can add your own TF2 logger (https://ray.readthedocs.io/en/latest/tune-usage.html#logging)
